api/src/main/java/jakarta/data/metamodel/NumericExpression.java

```diff
@@ -62,4 +66,17 @@ default NumericExpression<T,N> times(NumericExpression<T,N> other) {
     default NumericExpression<T,N> divide(NumericExpression<T,N> other) {
         return NumericOperatorExpression.of(DIVIDE, this, other);
     }
+
+    default NumericExpression<T,Long> asLong() {
```

Comment:
Does it make sense to include asInt as well?

Reply:
It might. I'm just not sure what we think about narrowing conversions...

Reply:
I think it would be difficult to commit to that, given the heterogeneity of the datastore we're targeting.